Refactor CI scripts to support more components (#473)
opea-project/GenAIInfra: helm-charts/CI.md

# CI guidelines for helm charts

## Table of Contents

- [Infra Setup](#infra-setup)
- [Add new test case](#add-new-test-case)

## Infra Setup

> Helm charts are orginized as:

Reviewer: ```suggestion
Helm charts are organized as:
```

1). E2E examples are directly located under helm-charts/ directory
2). Charts for components are located under helm-charts/common directory.

> We have Xeon servers with Gaudi accelerator card, so basicly we'll cover running workloads for cpu only and with Gaudi accelerated.For other configurations/feature coverage, it's case by case.

Reviewer: There's typo, but IMHO this can be simplified to:
```suggestion
We have Xeon servers with Gaudi accelerator card, which covers CPU and Gaudi accelerated workloads. For coverage of other configurations/features, it's case by case.
```

> We'll run CI test when detected values file beginning with "ci-" and ending with "values.yaml"(`ci-*values.yaml`).

Reviewer: Simpler e.g. as:
```suggestion
CI tests values files which names are listed in `ci-*values.yaml` files.
```

## Add new test case

Add a new values file following the naming rules: `ci-*values.yaml`.

> If the values file for CI is exactly the same with another one, you can just use a symbol link.

Reviewer: ```suggestion
If the values file for CI is identical to another one, a symbolic link can be used.
```
